[TC-8978] Add shipment polling (#65)
tradecloud/tradecloud-docs-api-v2: api/webhook-vs-polling.md

* `eventName:`The [eventName](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model"\) summarizes what has happened.
* `orderEvent`: The actual order event, see [OrderEvent](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model" and "OrderEvent"\) and [Receive order response](./).
* `orderDocumentsEvent`: Or the actual order documents event, see [OrderDocumentEvent](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-webhook-client/specs.yaml#/order-webhook%20endpoints/webhookPost) \(click "Model" and "OrderDocumentsEvent"\).

> * `shipmentEvent`: The actual shipment, see [Shipment](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment-webhook-connector/specs.yaml#/shipment-webhook%20endpoints/webhookPost) \(click "Model" and "Shipment"\).

Reviewer: "Or the actual shipment".
perhaps make this a sublist, to indicate that you will either get one of these 3 events, but never multiple at the same time.

Author: Created two lists

> Use `POST` or `PUT` when:

Reviewer: ```suggestion
Use `POST` when:
```

PUT is not supported by the `order-webhook-connector` or `shipment-webhook-connector`

Author: Removed PUT everywhere

* You want to receive real time order or shipment events.
* You want to receive the order event or shipment event content.

> * You only want to receive selected order or shipment events.

Reviewer: ```suggestion
* You only want to receive order or shipment events of a specific type.
```

{% hint style="info" %}
Pro's:

> * Semi real time, receive the order or shipment within seconds.

Reviewer: ```suggestion
* Real time, receive the order or shipment within a second.
```

This isn't any slower than `POST`, right?

* You can filter on which order or shipment events to receive, in the order & shipment webhook settings in your company profile.
* You do not have to build or configure the polling pattern.

Con's:

* You need to fetch the order or shipment.
* You cannot see what order or shipment event happened.
* You need to build or configure a webhook at your side.
* You need to publish the webhook on the internet \(web server and firewall required\).
* You need to obtain and configure a public SSL certificate.
{% endhint %}

## The polling pattern

Check if there are new or updated orders or shipments every polling period, typically 5 minutes, by using the last updated date time stamp of the last fetched order or shipment.

> The polling pattern is most suitable for companies with low volume orders, and not willing to invest in a web server, firewall and SSL certificate.

Reviewer: Nit: somehow I'm reading this quite negative "not willing to invest in..."

Author: That is what I mean, Hollandse Zuinigheid

* Not real time, a polling period is typically 5 mins.
* You need to build or configure a periodic polling pattern at your side.

> * You cannot filter on which order or shipments events to receive, you will receive any order line change.

Reviewer: ```suggestion
* You cannot filter on which order or shipments events to act on, you will receive any order line / shipment line change.
```

Author: Used `any order or shipment change` as lines are not relevant.

* You cannot see what order or shipment event happened, multiple events may have happened.
{% endhint %}

### Polling usage

#### Step 1. Fetch updated orders or shipments periodically

Fetch every polling period, typically 5 minutes, all orders or shipments which are new or changed since last date time.

* Use the latest `lastUpdatedAt` from previous poll request in the `lastUpdatedSince` filter.

> * Sorting is set automatically to `lastUpdatedAt` order `asc` \(latest `lastUpdatedAt` will be in the last order in the response body\)

Reviewer 2: I think the sentence below fitst better. 
(latest `lastUpdatedAt` will be last in the response body\

Author: Ah, I misread the `order` part. But `latest lastUpdatedAt will be` reallly `in the last order in the response body`
What means `will be last in the response body`?

Author: improved

[Search orders OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-search/specs.yaml#/order-search/searchRoute)
{% endhint %}

{% hint style="info" %}
[Search shipments OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment/specs.yaml#/shipment/searchShipmentsRoute)
{% endhint %}

#### Step 2. Process the orders in the search response body

See the [Search orders OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/order-search/specs.yaml#/order-search/searchRoute).

> * Use the `lastUpdatedAt` on order line level to filter on if the line has been changed.

Reviewer: ```suggestion
* Use the `lastUpdatedAt` on order line level to filter the lines that have been changed.
```

* Use the `status` field to filter on order process and logistics status.

> See the [Search shipments OpenAPI Specification](https://swagger-ui.accp.tradecloud1.com/?url=https://api.accp.tradecloud1.com/v2/shipment/specs.yaml#/shipment/searchShipmentsRoute).

Reviewer: Should this be 2 separate blocks? It feels a bit weird.
Perhaps link both API specs, and then provide intructions (instructions are the same, right?)

Author: Removed the embedded, duplicate API spec, it was something from the past, we now have real API specs

> * Use the `status` field to filter on shipment process and logistics status.

Reviewer 2: This is not yet implemented

Author: Removed

> #### Step 4. Store the `lastUpdatedAt` for the next polling request

Reviewer 2: Step 3 is missing.

Author: Renamed to `3`, not sure where `3` went
